barretenberg/cpp/pil/vm2/trees/written_public_data_slots_tree_check.pil

```diff
@@ -0,0 +1,132 @@
+include "merkle_check.pil";
+
+include "../ff_gt.pil";
+include "../poseidon2_hash.pil";
+include "../constants_gen.pil";
+include "../precomputed.pil";
+
+namespace written_public_data_slots_tree_check;
```

Comment:
Can we include a little bit of a write up here about this internal slots tree. The main nuances to highlight are probably: 1) this is tracking public tree slots only and not the values and 2) that this is a per-tx tree and has a lifetime that only persists during the tx